learn-data-structures-by-building-a-shopping-cart/index15.js

```diff
@@ -0,0 +1,59 @@
+const products = [
+  {
+    id: 0,
+    name: "Vanilla buttercream cupcake",
+    price: 2.99
+  },
+  {
+    id: 1,
+    name: "French Macaroon",
+    price: 3.99
+  },
+  {
+    id: 2,
+    name: "Fruit sprinkles cupcake",
+    price: 3.99
+  },
+  {
+    id: 3,
+    name: "Pink flower cupcake",
+    price: 5.99
+  }
+];
+
+class ShoppingCart {
+  constructor() {
+    this.items = [];
+  }
+
+  addItem(id, products) {
+    const item = products.find(item => item.id === id);
+    this.items.push(item);
+  }
+
+  getItems() {
+    return this.items;
+  }
+  
+  getCount() {
+    return this.items.length;
+  }
+  
+  removeItem(id) {
+    const index = this.items.indexOf(id);
+    this.items.splice(index, 1);
+  }
+  
+  clearCart() {
+    this.items = [];
+  }
+}
+
+/*
+It's mother's day and our store wants to give discounts to all of our products.
```

Comment:
It's Mother's Day and our store wants to discount all of our products.
In the `constructor`, make a new variable called `discountPercentage` and assign it the number 30 as in 30 percent.
Remember to use `this`.